tests/tests_geomstats/test_full_rank_correlation_matrices.py

```diff
@@ -1,148 +1,237 @@
-        result = self.bundle.tangent_riemannian_submersion(vertical, mat)
+from geomstats.geometry.symmetric_matrices import SymmetricMatrices
+from tests.conftest import TestCase, autograd_tf_and_torch_only
+from tests.data_generation import LevelSetTestData, TestData
+from tests.parametrizers import LevelSetParametrizer, Parametrizer
+
+
+class TestFullRankCorrelationMatrices(TestCase, metaclass=LevelSetParametrizer):
+
+    space = FullRankCorrelationMatrices
+    skip_test_extrinsic_intrinsic_composition = True
+    skip_test_intrinsic_extrinsic_composition = True
+
+    class TestDataRankFullRankCorrelationMatrices(LevelSetTestData):
+
+        n_list = random.sample(range(2, 5), 2)
+        space_args_list = [(n,) for n in n_list]
+        shape_list = [(n, n) for n in n_list]
+        n_samples_list = random.sample(range(2, 5), 2)
+        n_points_list = random.sample(range(2, 5), 2)
+        n_vecs_list = random.sample(range(2, 5), 2)
+
+        def random_point_belongs_data(self):
+            smoke_space_args_list = [(2,), (3,)]
+            smoke_n_points_list = [1, 2]
+            return self._random_point_belongs_data(
+                smoke_space_args_list,
+                smoke_n_points_list,
+                self.space_args_list,
+                self.n_points_list,
+            )
+
+        def projection_belongs_data(self):
+            return self._projection_belongs_data(
+                self.space_args_list, self.shape_list, self.n_samples_list
+            )
 
+        def to_tangent_is_tangent_data(self):
+            return self._to_tangent_is_tangent_data(
+                FullRankCorrelationMatrices,
+                self.space_args_list,
+                self.shape_list,
+                self.n_vecs_list,
+            )
+
+    testing_data = TestDataRankFullRankCorrelationMatrices()
+
+
+class TestCorrelationMatricesBundle(TestCase, metaclass=Parametrizer):
+    space = CorrelationMatricesBundle
+
+    class TestDataCorrelationMatricesBundle(TestData):
+        n_list = random.sample(range(2, 6), 3)
+        n_samples_list = random.sample(range(1, 7), 3)
+
+        def riemannian_submersion_belongs_to_base_data(self):
+            random_data = []
+            for n, n_samples in zip(self.n_list, self.n_samples_list):
+                bundle = CorrelationMatricesBundle(n)
+                point = bundle.base.random_point(n_samples)
+                random_data.append(dict(n=n, point=point))
+            return self.generate_tests([], random_data)
+
+        def lift_riemannian_submersion_composition_data(self):
+            random_data = []
+            for n, n_samples in zip(self.n_list, self.n_samples_list):
+                bundle = CorrelationMatricesBundle(n)
+                point = bundle.base.random_point(n_samples)
+                random_data.append(dict(n=n, point=point))
+            return self.generate_tests([], random_data)
+
+        def tangent_riemannian_submersion_data(self):
```

Comment:
ok

To add to design doc: this could be a property to test in a FiberBundleParametrizer